### PR TITLE
[STRATCONN] Expose DataDog stats client and tags in perform and performBatch handlers

### PR DESCRIPTION
**Pull request by brennan**
- This PR passes the DataDog stats client and DataDog tags to the `perform` and `performBatch` handler.
- The DataDog stats client and tags were passed to Actions in this prior `integrations` PR: https://github.com/segmentio/integrations/commit/cb50c4585e10eb8cbab5265b3a5a3e901ca9f471#diff-6db3e5ddc729acbbb968d5dd771dfa6d542245e980e9b9d12bdb2417d2e05163R205-R208

## Testing
- Testing completed successfully in stage. I deployed this branch/code to the staging environment and sent test events to validate that the desired metrics were collected: https://github.com/segmentio/action-destinations/blob/35c82a50874b8384a7a5504aa7b3c556178b4159/packages/destination-actions/src/destinations/google-analytics-4/pageView/index.ts#L49-L50

Screen Shot 2022-07-05 at 11 39 02 AM

https://segment.datadoghq.com/notebook/2808803/-testing-datadog-in-actions?live=false&start=1657045801741&range=467259

- [x] Added [unit tests](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing) for new functionality
- [x] Tested end-to-end using the [local server](https://github.com/segmentio/action-destinations/blob/main/docs/testing.md#local-end-to-end-testing)
- [x] [Segmenters] Tested in the staging environment

### CODE REVIEW

File: packages/destination-actions/src/destinations/snap-conversions-api/generated-types.ts [hunk 1]
```diff
@@ -10,7 +10,7 @@ export interface Settings {
    */
   snap_app_id?: string
   /**
-   * The unique ID assigned for a given application. It should be numeric for iOS, and the human interpretable string for Android. Required for app events.
+   * The unique ID assigned for a given application. It should be numeric for iOS, and the human interpretable string for Android. **Required for app events**.
```
Comment (marinhero) [inline packages/destination-actions/src/destinations/snap-conversions-api/generated-types.ts:13]: Want to remove this file from your PR? It seems unrelated to your PR.

File: packages/core/src/destination-kit/index.ts [hunk 2]
```diff
@@ -188,18 +197,14 @@ interface StatsClient {
   histogram(name: string, value?: number, tags?: string[]): void
 }
 
-interface StatsContext {
-  stats: StatsClient
+/** DataDog stats client and tags passed from the `CreateActionDestination`
+ * in the monoservice.
```
Comment (marinhero) [inline packages/core/src/destination-kit/index.ts:201]: Nit: consider linking to the line that does this.

**Review by marinhero — APPROVED** [review]
Well done! Left a couple of small non-blocking comments.

File: packages/core/src/destination-kit/index.ts (hunk 2 above)
Comment (brennan) [reply packages/core/src/destination-kit/index.ts:201]: I like this - will do.

File: packages/destination-actions/src/destinations/snap-conversions-api/generated-types.ts (hunk 1 above)
Comment (brennan) [reply packages/destination-actions/src/destinations/snap-conversions-api/generated-types.ts:13]: Just confirmed with @rvadera12 that it's safe to release this. It will have to go out eventually anyway.

**Review by dlasky — APPROVED** [review]
